### PR TITLE
feat: add x-components XModal

### PR DESCRIPTION
**Pull request by javieri-empathy**
EX-6701

Adds x-components xmodal

## Known issues

Because of how layouts are configured, you can't click on the overlay, as there are other layouts elements overlapping this element. So you are not able to close the XModal by clicking this element.

### CODE REVIEW

File: src/components/desktop/desktop.vue [hunk 1]
```diff
@@ -111,7 +111,7 @@
       BaseVariableColumnGrid,
       BaseIdModal,
       CartIcon,
-      Close: BaseEventsModalClose,
+      Close: CloseMainModal,
```
Comment (tajespasarela) [inline src/components/desktop/desktop.vue:114]: use the name `CloseMainModal` direclty instead of `Close`

File: src/components/mobile/mobile.vue [hunk 2]
```diff
@@ -112,7 +111,7 @@
       BaseIdModal,
       BaseIdModalOpen,
       BaseScroll,
-      Close: BaseEventsModalClose,
+      Close: CloseMainModal,
```
Comment (tajespasarela) [inline src/components/mobile/mobile.vue:114]: use the name `CloseMainModal` directly instead of `Close`

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by tajespasarela — APPROVED** [review]

File: rollup.config.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default createConfig({
   output: {
     chunkFileNames: chunkInfo => {
       switch (chunkInfo.name) {
-        case 'x-modal':
+        case 'custom-mamodal':
```
Comment (tiborux) [inline rollup.config.js:11]:
```suggestion
        case 'custom-main-modal':
```

Comment (tajespasarela) [inline rollup.config.js:11]: fix typo

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**